src/twisted/names/authority.py

```diff
@@ -270,141 +289,253 @@ def setupConfigNamespace(self):
+        """
+        This is a private API; do not use in application code.
+
+        Transform multiline statements into single lines.
+
+        @param lines: lines to work on
+        @type lines: iterable of L{bytes}
+
+        @return: iterable of continuous lines
+        """
+        l = []
         state = 0
         for line in lines:
             if state == 0:
-                if line.find('(') == -1:
-                    L.append(line)
+                if line.find(b'(') == -1:
+                    l.append(line)
                 else:
-                    L.append(line[:line.find('(')])
+                    l.append(line[:line.find(b'(')])
                     state = 1
             else:
-                if line.find(')') != -1:
-                    L[-1] += ' ' + line[:line.find(')')]
+                if line.find(b')') != -1:
+                    l[-1] += b' ' + line[:line.find(b')')]
                     state = 0
                 else:
-                    L[-1] += ' ' + line
-        lines = L
-        L = []
-        for line in lines:
-            L.append(line.split())
-        return filter(None, L)
+                    l[-1] += b' ' + line
+        return filter(None, (line.split() for line in l))
 
 
     def parseLines(self, lines):
-        TTL = 60 * 60 * 3
-        ORIGIN = self.origin
+        """
+        This is a private API; do not use in application code.
+
+        Parse C{lines}.
+
+        @param lines: lines to work on
+        @type lines: iterable of L{bytes}
+
+        @return: nothing
+        """
+        ttl = 60 * 60 * 3
+        origin = self.origin
 
         self.records = {}
 
-        for (line, index) in zip(lines, range(len(lines))):
-            if line[0] == '$TTL':
-                TTL = dns.str2time(line[1])
-            elif line[0] == '$ORIGIN':
-                ORIGIN = line[1]
-            elif line[0] == '$INCLUDE': # XXX - oh, fuck me
+        for line in lines:
+            if line[0] == b'$TTL':
+                ttl = dns.str2time(line[1])
+            elif line[0] == b'$ORIGIN':
+                origin = line[1]
+            elif line[0] == b'$INCLUDE':
                 raise NotImplementedError('$INCLUDE directive not implemented')
-            elif line[0] == '$GENERATE':
-                raise NotImplementedError('$GENERATE directive not implemented')
+            elif line[0] == b'$GENERATE':
+                raise NotImplementedError(
+                    '$GENERATE directive not implemented'
+                )
             else:
-                self.parseRecordLine(ORIGIN, TTL, line)
+                self.parseRecordLine(origin, ttl, line)
+
+        # If the origin changed, reflect that within the instance.
+        self.origin = origin
 
 
     def addRecord(self, owner, ttl, type, domain, cls, rdata):
-        if not domain.endswith('.'):
-            domain = domain + '.' + owner
+        """
+        This is a private API; do not use in application code.
```

Comment:
That seems like an unrelated question, but... txchecker-travis is passing :).

You already have a docstring here, so if you just delete the line `This is a private API; do not use in application code.` it should still be fine with it.

Am I misunderstanding the question?